static/app/views/issueList/utils/useFetchIssueTags.tsx

```diff
@@ -0,0 +1,303 @@
+import {getBuiltInTags} from 'sentry/stores/tagStore';
+import TeamStore from 'sentry/stores/teamStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import {
+  getIssueTitleFromType,
+  IssueCategory,
+  IssueType,
+  type Organization,
+  PriorityLevel,
+  type TagCollection,
+  type User,
+} from 'sentry/types';
+import {SEMVER_TAGS} from 'sentry/utils/discover/fields';
+import {FieldKey, FieldKind, IsFieldValues, ISSUE_FIELDS} from 'sentry/utils/fields';
+import {Dataset} from 'sentry/views/alerts/rules/metric/types';
+
+type UseFetchIssueTagsParams = {
+  org: Organization;
+  projectIds: string[];
+  enabled?: boolean;
+  end?: string;
+  keepPreviousData?: boolean;
+  start?: string;
+  statsPeriod?: string | null;
+  useCache?: boolean;
+};
+
+/**
+ * Fetches the tags from both the Errors and IssuePlatform dataset
+ * and combines them with builtin and predfined tags.
+ */
+export const useFetchIssueTags = ({
+  org,
+  projectIds,
+  keepPreviousData = false,
+  useCache = true,
+  enabled = true,
+  ...statsPeriodParams
+}: UseFetchIssueTagsParams) => {
+  const {teams} = useLegacyStore(TeamStore);
+  const {members} = useLegacyStore(MemberListStore);
+
+  const eventsTagsQuery = useFetchOrganizationTags(
+    {
+      orgSlug: org.slug,
+      projectIds,
+      dataset: Dataset.ERRORS,
+      useCache,
+      enabled,
+      keepPreviousData,
+      ...statsPeriodParams,
+    },
+    {}
+  );
+
+  const issuePlatformTagsQuery = useFetchOrganizationTags(
+    {
+      orgSlug: org.slug,
+      projectIds,
+      dataset: 'search_issues',
+      useCache,
+      enabled,
+      keepPreviousData,
+      ...statsPeriodParams,
+    },
+    {}
+  );
+
+  const userTeams = teams.filter(team => team.isMember).map(team => `#${team.slug}`);
+  const usernames: string[] = members.map(getUsername);
+  const nonMemberTeams = teams
+    .filter(team => !team.isMember)
+    .map(team => `#${team.slug}`);
+
+  const suggestedAssignees: string[] = [
+    'me',
+    'my_teams',
+    'none',
+    // New search builder only works with single value suggestions
+    ...(org.features.includes('issue-stream-search-query-builder')
+      ? []
+      : ['[me, my_teams, none]']),
+    ...userTeams,
+  ];
+  const assignedValues: SearchGroup[] | string[] = [
+    {
+      title: t('Suggested Values'),
+      type: 'header',
+      icon: <IconStar size="xs" />,
+      children: suggestedAssignees.map(convertToSearchItem),
+    },
+    {
+      title: t('All Values'),
+      type: 'header',
+      icon: <IconUser size="xs" />,
+      children: [
+        ...usernames.map(convertToSearchItem),
+        ...nonMemberTeams.map(convertToSearchItem),
+      ],
+    },
+  ];
+  const eventsTags = [
+    ...(eventsTagsQuery.data || []),
+    ...(issuePlatformTagsQuery.data || []),
+  ];
+  const eventsTagCollection = eventsTags.reduce<TagCollection>((acc, tag) => {
+    acc[tag.key] = {...tag, kind: FieldKind.TAG};
```

Comment:
Sorry, one _last_ comment! We use the `totalValues` number on the event tag to sort keys in the "Event Tags" section (higher values go to the top). If the same tag key gets returned from both datasets, this will override it with the issue platform response. If we come across a tag that's already in the map, we should merge the `totalValues` value